src/measure_theory/integral/average.lean

```diff
@@ -384,5 +537,157 @@ by simpa only [average_eq_integral]
   using exists_not_mem_null_average_le (is_probability_measure.ne_zero μ) hf hN
 
 end probability_measure
-end first_moment
+end first_moment_real
+
+section first_moment_ennreal
+variables {N : set α} {f : α → ℝ≥0∞}
+
+/-- **First moment method**. A measurable function is smaller than its mean on a set of positive
+measure. -/
+lemma measure_le_set_laverage_pos (hμ : μ s ≠ 0) (hμ₁ : μ s ≠ ∞)
+  (hf : ae_measurable f (μ.restrict s)) (hs : null_measurable_set s μ) :
```

Comment:
this one should be true assuming only `hs`.

Reply:
Haven't figured this one out. Do you have a hint?

Reply:
Sorry, I meant you just need `hf`. This one is wrong if you just assume `hs` (take `s = univ`. There exists a non-measurable function which is positive everywhere but with average zero).

Reply:
Sorry, I am still not seeing it. I would have thought of using `measure.le_restrict_apply` but the inequality goes the wrong way.

Reply:
See my suggestion below